### DISCUSSION TITLE
LINQ might get a "left join" operator in .net 10

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#125] [General]
# LINQ might get a "left join" operator in .net 10

LINQ has a Join operator, that basically translates to something like a SQL INNER JOIN. There is no built in operator for a LEFT JOIN in LINQ, but you can achieve the same result by using a combination of GroupJoin, SelectMany and DefaultIfEmpty. But there is a chance that this might change in the future, as there is a proposal to add a LeftJoin operator in .net 10.

https://steven-giesel.com/blogPost/82ebfd51-23c0-44b7-9602-628d1aba5f3c

**Comment by juniorgrao-evolve** [comment]
For some reason you omitted the <i><b>.Table1</b></i> in the proposal code right after your <i><b>_dbContext</b></i>. So I believe you would like to have as below:

var query = dbContext.Table1.LeftJoin(_dbContext.Table2, t1 => t1.T2Id, t2 => t2.Id, (t1, t2) => new { t1, t2 });

**Reply by linkdotnet** [reply]
You are absolutely right! Sorry that I missed that -.-

**Comment by TropinAlexey** [comment]
I prefer to use mixed syntax. Then code is more understandable and pretty much like SQL that you getting on output. And that .DefaultIfEmpty() makes your join LEFT

var documents = 
            from d in DbContext.Documents.Where(d => documentIds.Contains(d.Id))
            from dt in DbContext.DocumentTypes.Where(dt => dt.Id == d.TypeId).DefaultIfEmpty()
            select new DocumentDto
            {
                Id = d.Id,
                DocumentTypeId = d.TypeId,
                DocumentTypeName = dt.Name
            };

**Reply by linkdotnet** [reply]
That is a very nice approach I didn’t think of to be honest! I like it
